plugin.cpp

```diff
@@ -248,6 +253,55 @@ PLUGIN_HANDLE plugin_init(ConfigCategory* config,
 	return (PLUGIN_HANDLE)info;
 }
 
+
+/**
+ * Flatten nested datapoint
+ *
+ * @param datapoint	datapoint to flatten
+ * @param assetName	assetname to which datapoint belongs
+ * @param newReadings	vector for newreadings of flatten datapoint
+ */
+void flattenDatapoint(Datapoint *datapoint, std::string assetName, vector<Reading *>& newReadings)
+{
+	static std::string datapointName = datapoint->getName();
+	DatapointValue datapointValue = datapoint->getData();
+
+	std::vector<Datapoint *> *&nestedDP = datapointValue.getDpVec();
+
+	auto dpit = nestedDP->begin();
+	for (auto dpit = nestedDP->begin(); dpit != nestedDP->end(); dpit++)
+	{
+
+		std::string name = (*dpit)->getName();
+		DatapointValue &val = (*dpit)->getData();
+		if (val.getType() == DatapointValue::T_STRING)
```

Comment:
Rather than do this if you structure it as a set ion if's for all the ones that need different actions, e.g. the DP_DICT and DP_LIST type data points. Then a single else clause that deals with all the non-DP_DICT/DP_LIST datapoints, since the action is the same for all of them., it is more future proof as new datapoint types get added. It also has the bonus of making the code shorter and more clear.